packages/web-components/fast-foundation/src/toolbar/toolbar.ts

```diff
@@ -241,12 +242,15 @@ export class FASTToolbar extends FASTElement {
      * Set the activeIndex and focus the corresponding control.
      *
      * @param activeIndex - The new index to set
+     * @param enforceFocus - Whether to enforce the focus on the active element or not
      * @internal
      */
-    private setFocusedElement(activeIndex: number = this.activeIndex): void {
+    private setFocusedElement(activeIndex: number = this.activeIndex, enforceFocus: boolean = true): void {
         this.activeIndex = activeIndex;
         this.setFocusableElements();
-        this.focusableElements[this.activeIndex]?.focus();
+        if (enforceFocus) {
```

Comment:
What if instead of passing an "enforceFocus" param we instead did a check like
' if (this.contains(document.activeElement))'
which should only move focus if focus is currently in the toolbar?

Reply:
We do something similar in combobox -https://github.com/microsoft/fast/blob/9af1b251df57244d61524bf5af757839373249a0/packages/web-components/fast-foundation/src/combobox/combobox.ts#L339

Reply:
Good catch, btw.

Reply:
Thanks a lot for the review @scomea - your solution is much nicer.

Reply:
No worries, I did miss something though.  Can't simply check document.activeElement as that won't work right for toolbars in shadow dom so we should account for that.

Just copy this from data-grid-cell and get the active element from the function instead:

https://github.com/microsoft/fast/blob/961ebd203e9d2584940ca4849a65aa3ff3f781a1/packages/web-components/fast-foundation/src/data-grid/data-grid-cell.ts#L295C13-L295C33

We should probably consolidate that function into utilities and update combobox too (but not in this pr).

Reply:
https://github.com/microsoft/fast/pull/6899